src/lib/settings/group/SettingsArray.js

```diff
@@ -0,0 +1,155 @@
+const { resolveGuild, arraysStrictEquals, isNumber } = require('../../util/util');
+const Type = require('../../util/Type');
+
+const checkForIndex = (value) => Array.isArray(value) && value.length === 2 && typeof value[0] === 'number';
+
+class SettingsArray {
+
+	constructor(entry) {
+		Object.defineProperty(this, 'base', { value: null, writable: true });
+
+		// The Entry this SettingArray refers to
+		Object.defineProperty(this, 'entry', { value: entry });
+
+		this.data = entry.default;
+	}
+
+	get gateway() {
+		return this.base.gateway;
+	}
+
+	get(index) {
+		if (index !== undefined && index !== null) {
+			if (isNumber(index)) return this.data[index];
+			throw new TypeError(`The value for index must be of type 'number'. Got: ${new Type(index)}`);
+		}
+		return this.data.slice();
+	}
+
+	includes(value) {
+		return this.data.includes(value);
+	}
+
+	find(...args) {
+		return this.data.slice().find(...args);
```

Comment:
Any reason for the slice? `find` doesn't alter the array in place.. 
Don't we want to deepClone the result? Then I could see the use for slice, but in my opinion it'd be best to deepClone the result of find